### PR TITLE
Add http2_request example in sh2lib component

### PR DESCRIPTION
**Pull request by mahavirj**
This example depends on sh2lib abstraction layer for HTTP/2 protocol with TLS. Hence, its kept inside the sh2lib component itself.

Note: ESP-IDF version prior to 5.2 had this example as a part of it but it will be removed starting 5.2 release.

# Checklist

NA - Adding example inside an existing component

# Change description
_Please describe your change here_

### CODE REVIEW

File: sh2lib/examples/http2_request/pytest_http2_request.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    except Exception as msg:
+        logging.info('Exception occurred when connecting to {}: {}'.format(TEST_SERVER, msg))
+        return False
+    finally:
+        conn.close()
+
+
+@pytest.mark.esp32
+@pytest.mark.ethernet
+def test_examples_protocol_http2_request(dut: Dut) -> None:
```
Comment (hfudev) [inline sh2lib/examples/http2_request/pytest_http2_request.py:33]: from idf_build_apps.constants import IDF_VERSION_MAJOR

@pytest.mark.skipif(IDF_VERSION_MAJOR < 4, reason='does not support old releases')

Comment (hmalpani) [reply sh2lib/examples/http2_request/pytest_http2_request.py:33]: Or we can use `--collect-app-info` with `idf-build-apps` while building the examples and then use `app-info-filepattern` while running the examples. This is the way we do in ESP-IDF CI?

Comment (hfudev) [reply sh2lib/examples/http2_request/pytest_http2_request.py:33]: Also need to modify the `conftest.py`, which will make it more complicated. I don't recommend following the ESP-IDF CI in this repo, that's too complex.

Comment (mahavirj) [reply sh2lib/examples/http2_request/pytest_http2_request.py:36]: Thanks, added the change.

**Review by hmalpani — APPROVED** [review]

**Pull request merged**